src/Pate/Memory/MemTrace.hs

```diff
@@ -703,6 +715,63 @@ execMacawStmtExtension (MacawArchEvalFn archStmtFn) mkundef mvar globs stmt
       off' <- bvZext sym w offset
       pure (LLVMPointer region off')
 
+newtype MacawSyscallModel sym arch = MacawSyscallModel ()
```

Comment:
Could you add a note on what you envision this being in the long term? Might this be something like functions implementing the models for various system calls?